test/src/test/java/org/corfudb/runtime/object/CorfuSMRObjectProxyTest.java

```diff
@@ -260,4 +261,40 @@ public void canUseCustomSerializer() throws Exception {
         assertThat(test.get("a")).isEqualTo("b");
     }
 
+    /**
+     * Once a SMR map is created with a serializer, we should
+     * not be able to change the serializer by re-opening the map with
+     * a different serializer.
+     *
+     * @throws Exception
+     */
+    @Test
```

Comment:
What about the case where the serializer is set from the persisted data and not the open argument?